drud/ddev
# Add utilities and tests for CIFS simulated symlinks, add to ddev composer, fixes #1283

## The Problem/Issue/Bug:

OP #1283: On Win10 with docker-for-windows, the CIFS filesystem inside the container uses *simulated* XSym symlinks, which mostly work inside the container, are completely incomprehensible on the host, and don't *completely* work, especially on nontrivial TYPO3 projects (see OP for complete details).

Mostly this doesn't affect most Drupal projects, and doesn't affect plain-vanilla TYPO3 projects.

However, Windows 10 on the *host* has the capability of creating real, valid symlinks that are perfectly fine inside the container. But only if it's set to "developer" mode.

## How this PR Solves The Problem:

* Add simulated XSym symlink detection and replacement tools (and tests)
* Add to the "composer" commands a run through of replacing, if on docker-for-windows.

## Manual Testing Instructions:

* With a Windows 10 Pro system with developer mode, use `ddev composer install` or `ddev composer create` to create something that has symlinks. It should notify you of the creation of the symlinks.
* Without developer mode, try the same thing; it should warn about the issue.

## Automated Testing Overview:

* The utilities added have their own tests.
* The composer tests already do what's needed

Probably we should add a symlink to the repos used in the composer tests.

## Related Issue Link(s):

OP #1283 

## Release/Deployment notes:
